code/game/objects/items/blueprints.dm

```diff
@@ -101,7 +114,38 @@
 		if(get_area_type()!=AREA_STATION)
 			return
 		edit_area()
-	updateUsrDialog()
+	if(href_list["view_blueprints"])
+		set_viewer(usr, "<span class='notice'>You flip the blueprints over to view the complex information diagram.</span>")
+	if(href_list["hide_blueprints"])
+		clear_viewer(usr,"<span class='notice'>You flip the blueprints over to view the simple information diagram.</span>")
+	if(href_list["refresh"])
+		clear_viewer(usr)
+		set_viewer(usr)
+
+	attack_self(usr) //this is not the proper way, but neither of the old update procs work! it's too ancient and I'm tired shush.
+
+/obj/item/areaeditor/blueprints/proc/get_images(turf/T, viewsize)
+	. = list()
+	for(var/tt in RANGE_TURFS(viewsize, T))
+		var/turf/TT = tt
```

Comment:
oh no, spare us from the compile time delay, lets make our game perform poorly for any more than 25 players instead.